docs/content/guide/compiler.ngdoc

```diff
@@ -380,3 +380,106 @@ restrict: 'E',
 replace: true
 ```
 
+### Double Compilation, and how to avoid it
+
+Double compilation occurs when an already compiled part of the DOM gets compiled again. This is an
+undesired effect and can lead to misbehaving directives, performance issues, and memory
+leaks.
+A common scenario where this happens is a directive that calls `$compile` in a directive link
+function on the directive element. In the following **faulty example**, a directive adds a mouseover behavior
+to a button with `ngClick` on it:
+
+```
+angular.module('app').directive('addMouseover', function($compile) {
```

Comment:
I would still prefer a comment above this line; something like:

```js
// BAD - Don't do this
```

And an equivalent comment in the recommended alternatives; e.g.:

```js
// OK - You can do this
```

But I can leave without if you think it's too much. Up to you.